crates/primitives/digest-log/src/lib.rs

```diff
@@ -1,64 +1,59 @@
-// This file is part of Frontier.
-//
-// Copyright (c) 2020-2022 Parity Technologies (UK) Ltd.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-// 	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+//! Utility to read the Starknet block form the Substrate block digest
+//!
+//! Following the wrapper block model, each one of the madara block contains a starknet block.
+//! This block is not only stored in the chain storage, but also pushed inton the wrapper block
+//! itself.
+//!
+//! We expect the starknet pallet to push a log into the substrate digest in it's `on_finalize`
+//! hook. This log must contain the whole new starknet block.
+//!
+//! In the current state of this crate, only one single log must be pushed to the digest each block,
+//! and it should contain the starknet block. Pushing more log will make it impossible for this set
+//! of reader functions to operate properly.
 
 #![cfg_attr(not(feature = "std"), no_std)]
 #![allow(clippy::large_enum_variant)]
 #![deny(unused_crate_dependencies)]
 
+mod error;
+#[cfg(test)]
+mod tests;
+
+pub use error::FindLogError;
 use mp_starknet::block::Block as StarknetBlock;
 use scale_codec::{Decode, Encode};
-use sp_core::H256;
 use sp_runtime::generic::{Digest, OpaqueDigestItemId};
 use sp_runtime::ConsensusEngineId;
 
 pub const MADARA_ENGINE_ID: ConsensusEngineId = [b'm', b'a', b'd', b'a'];
 
-#[derive(Clone, PartialEq, Eq)]
+/// A Madara log
+///
+/// Right now we only expect Madara to log the Starknet block,
+/// but other usecases may appears later on.
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub enum Log {
-    Post(PostLog),
-}
-
-#[derive(Decode, Encode, Clone, PartialEq, Eq)]
-pub enum PostLog {
-    /// Ethereum block hash.
-    #[codec(index = 3)]
-    BlockHash(H256),
+    #[codec(index = 0)]
+    Block(StarknetBlock),
 }
 
-#[derive(Decode, Encode, Clone, PartialEq, Eq)]
-pub struct Hashes {
-    /// Starknet block hash.
-    pub block_hash: H256,
+/// Return the wrapped [StarknetBlock] contained in the [Digest]
+pub fn find_starknet_block(digest: &Digest) -> Result<StarknetBlock, FindLogError> {
+    find_log(digest).map(|log| match log {
+        Log::Block(b) => b,
+    })
 }
 
-impl Hashes {
-    pub fn from_block(block: StarknetBlock) -> Self {
-        Hashes { block_hash: block.header().hash() }
-    }
-}
-
-#[derive(Clone, Debug)]
-pub enum FindLogError {
-    NotFound,
-    MultipleLogs,
+/// Return the Madara [Log]
+pub fn find_log(digest: &Digest) -> Result<Log, FindLogError> {
```

Comment:
same, please conform to the format we use.
Example:
```rust
        /// The invoke transaction is the main transaction type used to invoke contract functions in
        /// Starknet.
        /// See `https://docs.starknet.io/documentation/architecture_and_concepts/Blocks/transactions/#invoke_transaction`.
        /// # Arguments
        ///
        /// * `origin` - The origin of the transaction.
        /// * `transaction` - The Starknet transaction.
        ///
        ///  # Returns
        ///
        /// * `DispatchResult` - The result of the transaction.
        ///
        /// # TODO
        /// * Compute weight
        #[pallet::call_index(1)]
        #[pallet::weight(0)]
        pub fn invoke(_origin: OriginFor<T>, mut transaction: Transaction) -> DispatchResult 
    ```

Reply:
La doc décrit déjà le return:
`/// Return the Madara [Log]`

Et la doc du crate explique assez clairement la logique à l'oeuvre dans ce crate:
```
//! Utility to read the Starknet block form the Substrate block digest
//!
//! Following the wrapper block model, each one of the madara block contains a starknet block.
//! This block is not only stored in the chain storage, but also pushed inton the wrapper block
//! itself.
//!
//! We expect the starknet pallet to push a log into the substrate digest in it's `on_finalize`
//! hook. This log must contain the whole new starknet block.
//!
//! In the current state of this crate, only one single log must be pushed to the digest each block,
//! and it should contain the starknet block. Pushing more log will make it impossible for this set
//! of reader functions to operate properly.
```

Est-ce que tu crois qu'il y a vraiment besoin de rajouter une ligne qui dit:
```
# Arguments
digest: A Substrate block [Digest]
```

Reply:
A la limite je peux enrichir la phrase déjà présente
`/// Return the Madara [Log] contained in a Substrate block [Digest]`
Pour être vraiment archi explicite.